### PR TITLE
[Pending #215] Enforce entity checks on most commands

### LINKED ISSUE #137: [Alek] `screen [invalid enitity]` error is not obvious enough
- `screen contact 1` does not return a specific error, it only reply with usage. 
- perhaps it is better to display an explicit error message like "invalid entity type!!!!!!!!!!!!!!!"

image

### LINKED ISSUE #150: [Alek] `delete [invalid entity]` error not specific
- index is present, entity is missing
image

- where exactly? should I replace the index with contact?
image

### LINKED ISSUE #156: [General] unexpected add usage message
image

### LINKED ISSUE #169: [General] Standardize parser validation
- #137
- #139 
- #148 
- #150

### PR DESCRIPTION
**Pull request by KengHian**
- close #150
- close #137
- close #169
- close #156
- Check if operation allowed
  -  
- Check if entity is valid
  -  
- Note
  - Not implemented in `Edit` & `Find` as they currently do not check entity, which is inconsistent with the command structure we are using. We should modify them, alternatively we could just disable them. 
  - I have not implement this in `ViewCompanyCommand` because it currently does not check entity?
  - `Match` & `Unmatch` do not take in entity so it is not needed

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -37,10 +37,29 @@ public static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Pref
...
+     * Throws ParseException if {@code entity} provided is not a valid entity in the address book.
+     * Leading and trailing whitespaces will be trimmed.
+     */
+    public static void requireValidEntity(String entity) throws ParseException {
+        requireNonNull(entity);
+        String trimmedEntity = entity.trim();
+
+        // TODO: Not sure if this is magic string?
+        switch (trimmedEntity) {
+        case "contact", "company", "job", "all":
```
Comment (Green-Tea-123) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:50]: If you don't wanna have the magic strings, we can just store the entity words into Person, Job, and Company as constants.

**Review by Green-Tea-123 — APPROVED** [review]
LGTM.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 1 above)
Comment (KengHian) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:50]: I guess that is fine, but does that mean we should change the entity word of all commands as well?

Also if we are going by this, I am not exactly sure where to store "all".

Comment (Green-Tea-123) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:50]: hmm yeah not sure where to store that lol

**Review by wassilililily — APPROVED** [review]
LGTM

**Pull request merged**